lib/datadog/appsec/contrib/rack/request_middleware.rb

```diff
@@ -49,6 +54,50 @@ def call(env)
             AppSec::Event.record(*both_response.map { |_action, event| event }) if both_response.any?
 
             request_return
+          ensure
+            add_waf_runtime_tags(context)
```

Comment:
‼️ ‼️ Important: I think there's a bug here -- if `@processor.ready? == false` in line 24, then this will be called with `nil` and the `add_waf_runtime_tags(nil)` will break the app.

If my assessment is correct, in addition to fixing it, I would also request adding test coverage for this :)

Reply:
Correct. It can also be the case if `@processor.new_context` fails.

Reply:
Okay, so I tried adding a spec for that and turns out it's a rabbit hole and derailing the PR. I'd rather address the spec aspect (especially those that run under contrib and appraisals) more generally by landing a proper architecture for that instead of hacking things around, which would be worse.

Reply:
> I'd rather address the spec aspect (especially those that run under contrib and appraisals) more generally by landing a proper architecture for that instead of hacking things around, which would be worse.

Even without specs, I think we should still guard around this case, to avoid calling `add_waf_runtime_tags(nil)`

Reply:
Haha, but I did! Just not at this call site in `ensure` but within `add_waf_runtime_tags`:

```
          def add_waf_runtime_tags(context)
            return unless active_trace
            return unless context            # <---- there
```

Reply:
👍  I would actually suggest going one step further and also doing it in the `ensure` as well, because this subtlety might not be obvious to the next person that touches this code, which may add another call to another method and forget to do the check.

`add_waf_runtime_tags(context) if context` would make it clear that `context` can be `nil`.

Reply:
Fair enough